src/message_interface/message_handler.rs

```diff
@@ -330,6 +330,35 @@ impl WalletMessageHandler {
                 })
                 .await
             }
+            #[cfg(feature = "participation")]
+            Message::GetParticipationEvent(event_id) => {
+                convert_async_panics(|| async {
+                    let event = self
+                        .account_manager
+                        .get_participation_event(event_id)
+                        .await?
+                        .ok_or_else(|| crate::Error::Storage(format!("event {event_id} not found")))?
```

Comment:
Feels a bit odd to return an error, I would put an option in the Response, we do that for some other types